redash/query_runner/snowflake.py

```diff
@@ -45,8 +44,9 @@ def configuration_schema(cls):
                 "warehouse": {"type": "string"},
                 "database": {"type": "string"},
                 "region": {"type": "string", "default": "us-west"},
+                "host": {"type": "string"},
```

Comment:
Can we make it an "extra" option? We added this option at some point, which will hide less frequently used settings under "Additional Settings" toggle when setting up the data source?

Here's an example for Postgres:
![CleanShot 2021-05-05 at 16 53 11@2x](https://user-images.githubusercontent.com/71468/117223428-7bf75700-adc2-11eb-931f-ac405e5f8adc.png)

And how it looks like in code:
https://github.com/getredash/redash/blob/9c8c1bfa9ade90eb147bfcda75e5f9e95b691e42/redash/query_runner/pg.py#L173-L178

Reply:
Done. I checked locally and it looks this way now:

![image](https://user-images.githubusercontent.com/69909379/117224997-f7382780-ae19-11eb-9c05-475988972736.png)

I also took the liberty to move the region under the account, as it's a better fit UX-wise IMO.

I also think that some of the logic in this component can be improved. I'll msg you about it, and can perhaps also do the changes.